### PR TITLE
[Sage-502] Sage 3 - Copy Button

### LINKED ISSUE #502: Update Copy button
## Description

Update component to match latest specs in Sage 3.0. The following were noted during initial audit:

- Implement option with a field/value
- Update JS settings to allow for including or excluding prefix and suffix values for different use cases

**Links**
- [Figma specs](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/WIP-Sage-3-%E2%80%94-Admin-Components?node-id=4195%3A20283)
- [Existing component(s)](https://sage-design-system.kajabi.com/pages/component/copy_text)

### PR DESCRIPTION
**Pull request by anechol**
## Description

Updates styling to Sage 3 specs:
- Updates padding and style to match Form Inputs
- Changes focus color to `Primary`
- Increases margin spacing between icon and text box

Figma file has also been added to the docs.

[Figma](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/Sage-3-for-Admin?node-id=4195%3A20283)

## Screenshots

|  Before  |  After  |
|--------|--------|
|||
Screen Shot 2021-09-30 at 9 59 41 AM|Screen Shot 2021-09-30 at 9 57 24 AM
Screen Shot 2021-09-30 at 10 00 41 AM|Screen Shot 2021-09-30 at 10 01 21 AM

### Rails
- View [Rails component](http://localhost:4000/pages/component/copy_text)
- Check that component styling updates are there

### React
- View [Storybook](http://localhost:4100/?path=/docs/sage-copytext--default)
- Check that component styling updates are there

## Testing in `kajabi-products`

(**LOW**) Updates Copy Text component to Sage 3 styling specs.

## Related

Closes #502

### CODE REVIEW

File: packages/sage-assets/lib/stylesheets/components/_copy_text.scss [hunk 1]
```diff
@@ -52,7 +49,7 @@ $-copy-text-color-hover: sage-color(charcoal, 500);
 .sage-copy-btn {
   @include sage-button-style-reset();
   @include sage-focus-outline();
-  @include sage-focus-outline--update-color($-copy-text-color-hover);
+  @include sage-focus-outline--update-color(sage-color(primary));
```
Comment (philschanely) [inline packages/sage-assets/lib/stylesheets/components/_copy_text.scss:52]: Should this be one of our variables form the top of the file?

File: packages/sage-assets/lib/stylesheets/components/_copy_text.scss [hunk 2]
```diff
@@ -15,17 +15,14 @@ $-copy-text-color-hover: sage-color(charcoal, 500);
 
   display: inline-block;
   max-width: 100%;
-  padding: 0 sage-spacing(xs);
 
   .sage-copy-btn & {
-    transition: map-get($sage-transitions, default);
-    transition-property: color, background-color;
+    transition: border 0.2s ease-in-out;
```
Comment (philschanely) [inline packages/sage-assets/lib/stylesheets/components/_copy_text.scss:20]: Would this be beneficial to capture in the `$sage-transitions` global var?

**Review by philschanely — APPROVED** [review]
I recommend bridging and checking out some existing uses to ensure there aren't unintended layout side-effects (if you haven't already 😊 ). Custom domain flow and some areas of CEDS come to mind. I agree the impact should be LOW but worth peaking just in case.

File: docs/app/helpers/components_helper.rb [hunk 3]
```diff
@@ -157,7 +157,7 @@ def sage_components
         react: "done",
         a11y: "done",
         react_component_slug: "sage-copytext--default",
-        figma_embed: "",
+        figma_embed: "https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2F9Km09NjlZHYWsMP7EGT8tI%2FSage-3-for-Admin%3Fnode-id%3D4195%253A20283",
```
Comment (monicawheeler) [inline docs/app/helpers/components_helper.rb:160]: 🙌

Comment (FuturaExtraBold) [reply docs/app/helpers/components_helper.rb:160]: 🔥

File: packages/sage-assets/lib/stylesheets/components/_copy_text.scss (hunk 2 above)
Comment (anechol) [reply packages/sage-assets/lib/stylesheets/components/_copy_text.scss:20]: Ah good catch! I've added this as an option to the variable

File: packages/sage-assets/lib/stylesheets/components/_copy_text.scss (hunk 1 above)
Comment (anechol) [reply packages/sage-assets/lib/stylesheets/components/_copy_text.scss:52]: Added this as a variable!

**Review by monicawheeler — APPROVED** [review]

**Pull request merged**